dlrs/main/classes/RollupServiceTest5.cls

```diff
@@ -133,6 +133,8 @@ private class RollupServiceTest5 {
     rollupSummaryOpp.CalculationMode__c = 'Scheduled';
     insert rollupSummaryOpp;
 
+    RollupSummariesSelector.summariesCacheByKey = new Map<String, List<RollupSummary>>();
```

Comment:
I am surprised to find the necessity to reset the @testvisible cache here but not in other test methods.

Reply:
The first time you cause a rollup to be executed during a test the cache is built. For a few tests we run once then insert a new Rollup record and cause the rollups to be executed again. In those cases the cached data doesn't hold the new config records and thus we have to invalidate the cache before proceeding.
This scenarios should never be an issue in production execution because we shouldn't be modifying rollup configs as a product of a transaction that executes a rollup. (At least I've never heard of it happening)
If we thought it might be a concern. We already have a trigger on the custom objects, we could add to that trigger logic to invalidate the cache after insert/update.

I didn't do anything about it because the custom object is a relic that should be deprecated and removed rather than reinforced and nobody could update the CMDT real-time as part of a triggered action.